gen3/bin/healthcheck.sh

```diff
@@ -0,0 +1,118 @@
+    local statusLine=$(g3kubectl describe pod $(echo $pod | jq -r '.name') | grep "Status:" -m 1)
+    if [[ "$(echo $statusLine | awk '{ print $2 }')" == "Terminating" ]]; then
+      # check how long it's been terminating
+      local s=$(echo $statusLine | grep -oP "\d+(?=s)")
+      local m=$(echo $statusLine | grep -oP "\d+(?=m)")
+      local h=$(echo $statusLine | grep -oP "\d+(?=h)")
+      if [[ $m -gt 10 || $s -gt 300 || $h -gt 0 ]]; then
+        terminatingPods=$(echo $terminatingPods | jq -r ". += [$pod]")
+      fi
+    fi
+  done <<< "$(echo $allPods | jq -c '.[]')"
+
+  # check status of nodes
+  # https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#nodestatus-v1-core
+  local allNodes=$(g3kubectl get nodes -o json | \
+    jq -r '[
+      .items[] | {
+        name: .metadata.labels."kubernetes.io/hostname", conditions: [ .status.conditions[] | { (.type): .status }] | add
+      }
+    ]')
+  local notReadyNodes=$(echo $allNodes | jq -r '.[] | select(.conditions.Ready != "True")')
+
+  # check internet access
+  local statusCode=$(curl -s -o /dev/null -I -w "%{http_code}" https://www.google.com)
+  local internetAccess=true
+  if [[ $statusCode -lt 200 || $statusCode -ge 400 ]]; then
+    internetAccess=false
+  fi
+
+  local healthy=true
+  if [[ "$internetAccess" = false || ! -z "$pendingLongPods" || ! -z "$terminatingPods" || ! -z "$unknownPods" || ! -z "$crashLoopPods" || ! -z "$evictedPods" || ! -z "$notReadyNodes" ]]; then
+    healthy=false
+  fi
+
+  local healthJson=$(echo '{}' | jq -r "{
+    pendingTimeoutPods: $pendingLongPods,
+    terminatingTimeoutPods: $terminatingPods,
+    unknownPods: [$unknownPods],
+    crashLoopBackOffPods: [$crashLoopPods],
+    evictedPods: [$evictedPods],
+    notReadyNodes: [$notReadyNodes],
+    internetAccess: $internetAccess
+  }")
+  echo $healthJson | jq -r '.'
+
+  if [[ "$HEALTH_SEND_SLACK" = true && "$healthy" = false ]]; then
+    if [[ "${slackWebHook}" == 'None' || -z "${slackWebHook}" ]]; then
+      slackWebHook=$(g3kubectl get configmap global -o jsonpath={.data.slack_webhook})
+    fi
+    if [[ "${slackWebHook}" == 'None' || -z "${slackWebHook}" ]]; then
+      echo "WARNING: slackWebHook is None or doesn't exist; not sending results to Slack"
+    else
+      local tmpHostname=$(g3kubectl get configmap manifest-global -o jsonpath={.data.hostname})
+      local formattedAttachment="{\"title\": \"Status\", \"text\": \"$(echo $healthJson | sed s/\"/\\\\\"/g | sed s/,/,\\n/g)\", \"color\": \"#FF0000\"}"
+      curl -X POST --data-urlencode "payload={\"text\": \"Healthcheck FAILED for ${tmpHostname}\n\", \"attachments\": [$formattedAttachment]}" "${slackWebHook}"
+    fi
+  fi
+}
+
+HEALTH_SEND_SLACK=false
+while [[ $# -gt 0 ]]; do
+  key="$1"
+  case $key in
+    '--slack')
+      HEALTH_SEND_SLACK=true
+      shift
+      ;;
+    *)
+      echo "Unrecognized flag"
+      help
+      exit 1
+      ;;
+  esac
+done
+
+gen3_healthcheck
```

Comment:
You can pass command line arguments through to gen3_healthcheck:
    `gen3_healthcheck "$@"`
, then move the argument checking into `gen3_healthcheck` rather than setting the global `HEALTH_SEND_SLACK` variable